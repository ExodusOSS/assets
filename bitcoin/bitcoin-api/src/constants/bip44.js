// back compatibility. Use @exodus/bip44-constants/by-ticker instead

export { default } from '@exodus/bip44-constants/by-ticker.js'
