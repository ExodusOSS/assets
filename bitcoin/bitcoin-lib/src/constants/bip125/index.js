// Marks TX input as Final although this doesn't guarantee
// the transaction wouldn't be replaceable
export const FINAL_SEQUENCE = 0xff_ff_ff_ff
// https://github.com/bitcoin/bitcoin/pull/6871/files#diff-7ec3c68a81efff79b6ca22ac1f1eabbaR859
// Signals Transaction is Replaceable
export const RBF_SEQUENCE = FINAL_SEQUENCE - 2

// Enables RBF support for these assets.
export const RBF_ASSET_NAMES = ['bitcoin']

export const RBF_FEE_MULTIPLIER = 0.7
export const RBF_FEE_BUMP_MULTIPLIER = 1.3
