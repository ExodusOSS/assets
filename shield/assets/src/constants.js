export const CT_UPDATEABLE_PROPERTIES = [
  'displayName',
  'displayTicker',
  'gradientColors',
  'gradientCoords',
  'icon',
  'info.description',
  'info.reddit',
  'info.telegram',
  'info.twitter',
  'info.website',
  'lifecycleStatus',
  'primaryColor',
  'properName',
  'properTicker',
]

export const CT_STATUS = {
  VERIFIED: 'v',
  CURATED: 'c',
  UNVERIFIED: 'u',
  DISABLED: 'd',
}

export const CT_ALL_STATUSES = Object.values(CT_STATUS)

export const CT_DEFAULT_SERVER = 'https://ctr.a.exodus.io/registry'
