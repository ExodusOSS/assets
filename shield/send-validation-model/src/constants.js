export const VALIDATION_TYPES = {
  ERROR: 0,
  WARN: 1,
  INFO: 2,
}
export const PRIORITY_LEVELS = {
  BASE: 0,
  MIDDLE: 1,
  TOP: 2,
}

export const FIELDS = {
  ADDRESS: 'address',
  AMOUNT: 'amount',
  MEMO: 'memo',
}
