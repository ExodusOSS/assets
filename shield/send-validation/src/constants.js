import sendValidationModel from '@exodus/send-validation-model'

export const { VALIDATION_TYPES, PRIORITY_LEVELS, FIELDS } = sendValidationModel

// TODO move each custom validation to its plugin
export const CAN_SEND_ZERO_AMOUNT = ['algorand']
export const ASSETS_WITHOUT_SELF_SEND = ['eosio', 'hedera', 'ripple', 'tronmainnet']
