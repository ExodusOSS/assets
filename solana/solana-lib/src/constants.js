import { PublicKey, StakeProgram, SystemProgram } from './vendor/index.js'

export const SPL_ASSOCIATED_TOKEN_ACCOUNT_PROGRAM_ID = new PublicKey(
  'ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL'
)
export const SYSTEM_PROGRAM_ID = SystemProgram.programId

export const STAKE_PROGRAM_ID = StakeProgram.programId

export const TOKEN_PROGRAM_ID = new PublicKey('TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA')

export const TOKEN_2022_PROGRAM_ID = new PublicKey('TokenzQdBNbLqP5VEhdkAS6EPFLC1PHnBqCXEpPxuEb')

export const MEMO_PROGRAM_ID = new PublicKey('MemoSq4gqABAXKb96qnH8TysNcWxMyWCqXgDLGmfcHr')

export const MAGIC_EDEN_ESCROW_PROGRAM_ID = new PublicKey(
  'MEisE1HzehtrDpAAT8PnLHjpSSkRYakotTuJRPjTpo8'
)

export const MPL_TOKEN_METADATA_PROGRAM_ID = new PublicKey(
  'metaqbxxUerdq28cj1RbAWkYQm3ybzjb6a8bt518x1s'
)

export const SYSVAR_INSTRUCTIONS_PUBKEY = new PublicKey(
  'Sysvar1nstructions1111111111111111111111111'
)

export const SEED = 'stake:0'

export const LAMPORTS_PER_SOL = 1_000_000_000

export const SOL_DECIMAL = Math.log10(LAMPORTS_PER_SOL)

export const SUPPORTED_TRANSACTION_VERSIONS = new Set(['legacy', 0])
