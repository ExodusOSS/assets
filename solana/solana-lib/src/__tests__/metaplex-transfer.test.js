import { MPL_TOKEN_METADATA_PROGRAM_ID, TOKEN_2022_PROGRAM_ID } from '../constants.js'
import {
  createMetaplexTransferTransaction,
  encodeData,
  TOKEN_STANDARD,
} from '../helpers/metaplex-transfer.js'

const RECENT_BLOCKHASH = 'A5EotUB6U3m5FQ7H5anvz9ZLZYjLLm7EMYEZXYLmAWHH'

const PROGRAMMABLE_NFT_TEST = {
  from: '9qPGTK92cRkfXzPVVEgz1bUYweNBY9PvoeoiavkMKmMs',
  to: '3LtDVz2HYAXakiZhv5pi9cetZXoWGutGjc8g4bQnRmWb',
  tokenMintAddress: '3YYjPyjkiV8KkiWNJV5SMdDJDjaTj8VZiJeuekMhVd7e',
  tokenStandard: TOKEN_STANDARD.ProgrammableNonFungible,
}

const NORMAL_NFT_TEST = {
  fromOwner: '9qPGTK92cRkfXzPVVEgz1bUYweNBY9PvoeoiavkMKmMs',
  toOwner: '3LtDVz2HYAXakiZhv5pi9cetZXoWGutGjc8g4bQnRmWb',
  mint: '3h9z827XPq51UyrV2WiD4Bx3Ku7W6RYbeyPzvZttrz8E',
}

test('Solana programmable NFT transaction', () => {
  const tx = createMetaplexTransferTransaction({
    ...PROGRAMMABLE_NFT_TEST,
    recentBlockhash: RECENT_BLOCKHASH,
  })

  expect(tx.instructions.length).toEqual(1)
  expect(tx.instructions[0].keys.length).toEqual(17)
  expect(tx.instructions[0].keys[7].pubkey.toBase58()).not.toEqual(
    MPL_TOKEN_METADATA_PROGRAM_ID.toBase58()
  )
  expect(tx.instructions[0].keys[8].pubkey.toBase58()).not.toEqual(
    MPL_TOKEN_METADATA_PROGRAM_ID.toBase58()
  )
  expect(tx.instructions[0].keys).toMatchSnapshot()
})

test('Solana programmable NFT transaction using token 2022', () => {
  const tx = createMetaplexTransferTransaction({
    ...PROGRAMMABLE_NFT_TEST,
    tokenProgram: TOKEN_2022_PROGRAM_ID.toBase58(),
    recentBlockhash: RECENT_BLOCKHASH,
  })

  expect(tx.instructions.length).toEqual(1)
  expect(tx.instructions[0].keys.length).toEqual(17)
  expect(tx.instructions[0].keys[13].pubkey.toBase58()).toEqual(TOKEN_2022_PROGRAM_ID.toBase58())
  expect(tx.instructions[0].keys).toMatchSnapshot()
})

test('Solana normal NFT transaction using metaplex transfer', () => {
  const tx = createMetaplexTransferTransaction({
    ...NORMAL_NFT_TEST,
    recentBlockhash: RECENT_BLOCKHASH,
  })

  expect(tx.instructions.length).toEqual(1)
  expect(tx.instructions[0].keys.length).toEqual(17)
  expect(tx.instructions[0].keys[7].pubkey.toBase58()).toEqual(
    MPL_TOKEN_METADATA_PROGRAM_ID.toBase58()
  )
  expect(tx.instructions[0].keys[8].pubkey.toBase58()).toEqual(
    MPL_TOKEN_METADATA_PROGRAM_ID.toBase58()
  )
  expect(tx.instructions[0].keys).toMatchSnapshot()
})

test('Solana metaplex transfer data', () => {
  const expectedBuffer = Buffer.from('3100010000000000000000', 'hex')

  const myBuffer = encodeData({ amount: 1 })

  expect(Buffer.compare(myBuffer, expectedBuffer)).toEqual(0)
})
