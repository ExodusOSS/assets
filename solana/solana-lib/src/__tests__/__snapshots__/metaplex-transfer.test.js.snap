exports[`Solana normal NFT transaction using metaplex transfer 1`] = `
[
  {
    "isSigner": false,
    "isWritable": true,
    "pubkey": "2a8MS8dWyyYNgBHgtzeTwrsDKsE6RnCoUqnonB4C8Xc3",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "11111111111111111111111111111111",
  },
  {
    "isSigner": false,
    "isWritable": true,
    "pubkey": "2a8MS8dWyyYNgBHgtzeTwrsDKsE6RnCoUqnonB4C8Xc3",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "11111111111111111111111111111111",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "11111111111111111111111111111111",
  },
  {
    "isSigner": false,
    "isWritable": true,
    "pubkey": "3s4yvESqATKTm6KGKoQhGaD4d8HAjq7XkKu7ACaFg3rs",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "BB11bFyjG8s4DwbB8rQvsekuaD3j5nZL2ojuATRg3N6x",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "metaqbxxUerdq28cj1RbAWkYQm3ybzjb6a8bt518x1s",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "metaqbxxUerdq28cj1RbAWkYQm3ybzjb6a8bt518x1s",
  },
  {
    "isSigner": true,
    "isWritable": false,
    "pubkey": "11111111111111111111111111111111",
  },
  {
    "isSigner": true,
    "isWritable": true,
    "pubkey": "11111111111111111111111111111111",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "11111111111111111111111111111111",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "Sysvar1nstructions1111111111111111111111111",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "auth9SigNpDKz4sJJ1DfCTuZrZNSAgh9sFD3rboVmgg",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "metaqbxxUerdq28cj1RbAWkYQm3ybzjb6a8bt518x1s",
  },
]
`;

exports[`Solana programmable NFT transaction 1`] = `
[
  {
    "isSigner": false,
    "isWritable": true,
    "pubkey": "8T3zM3cpo44PruakM9RxTET5JkTRByV6oxuZX8vAJpij",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "9qPGTK92cRkfXzPVVEgz1bUYweNBY9PvoeoiavkMKmMs",
  },
  {
    "isSigner": false,
    "isWritable": true,
    "pubkey": "D6ETjcNWVyejGvCDykoGxc3HBzNGHEEhac95xhMSdDa4",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "3LtDVz2HYAXakiZhv5pi9cetZXoWGutGjc8g4bQnRmWb",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "3YYjPyjkiV8KkiWNJV5SMdDJDjaTj8VZiJeuekMhVd7e",
  },
  {
    "isSigner": false,
    "isWritable": true,
    "pubkey": "4tX3ozaCeJy2ZY9o84BDYMNojrtwAe28xvQChcCWwig6",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "E6wcjEeDTTtratFjisEMadrTZ3yUPrWWqBwQRXpmkXfh",
  },
  {
    "isSigner": false,
    "isWritable": true,
    "pubkey": "vpYHzSJdgc6iXgU85cQ8w6fe3YAeGCYG3z4AMpVf6dZ",
  },
  {
    "isSigner": false,
    "isWritable": true,
    "pubkey": "FySESpdLQaCFQXvKVskfoDEFnDufN2qvBvnVGKUnH8ta",
  },
  {
    "isSigner": true,
    "isWritable": false,
    "pubkey": "9qPGTK92cRkfXzPVVEgz1bUYweNBY9PvoeoiavkMKmMs",
  },
  {
    "isSigner": true,
    "isWritable": true,
    "pubkey": "9qPGTK92cRkfXzPVVEgz1bUYweNBY9PvoeoiavkMKmMs",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "11111111111111111111111111111111",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "Sysvar1nstructions1111111111111111111111111",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "auth9SigNpDKz4sJJ1DfCTuZrZNSAgh9sFD3rboVmgg",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "metaqbxxUerdq28cj1RbAWkYQm3ybzjb6a8bt518x1s",
  },
]
`;

exports[`Solana programmable NFT transaction using token 2022 1`] = `
[
  {
    "isSigner": false,
    "isWritable": true,
    "pubkey": "5qYQ3n4pYYDBsSyx6WK3g8mGPFH4G23TzrG3ssXFTVVF",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "9qPGTK92cRkfXzPVVEgz1bUYweNBY9PvoeoiavkMKmMs",
  },
  {
    "isSigner": false,
    "isWritable": true,
    "pubkey": "Bh6Utbgqtdyw9QUHQzBi6vkyiw9VU62a7Z8At4sooiBT",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "3LtDVz2HYAXakiZhv5pi9cetZXoWGutGjc8g4bQnRmWb",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "3YYjPyjkiV8KkiWNJV5SMdDJDjaTj8VZiJeuekMhVd7e",
  },
  {
    "isSigner": false,
    "isWritable": true,
    "pubkey": "4tX3ozaCeJy2ZY9o84BDYMNojrtwAe28xvQChcCWwig6",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "E6wcjEeDTTtratFjisEMadrTZ3yUPrWWqBwQRXpmkXfh",
  },
  {
    "isSigner": false,
    "isWritable": true,
    "pubkey": "5yLVx4aqhi1B8d6dnywQNzowuFgrFSHXajKmGiDpe7yG",
  },
  {
    "isSigner": false,
    "isWritable": true,
    "pubkey": "ALbiLTmXFG3WvWXmoHNCWfnrnGoNsoiZbZ2Vt5YpWbbX",
  },
  {
    "isSigner": true,
    "isWritable": false,
    "pubkey": "9qPGTK92cRkfXzPVVEgz1bUYweNBY9PvoeoiavkMKmMs",
  },
  {
    "isSigner": true,
    "isWritable": true,
    "pubkey": "9qPGTK92cRkfXzPVVEgz1bUYweNBY9PvoeoiavkMKmMs",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "11111111111111111111111111111111",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "Sysvar1nstructions1111111111111111111111111",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "TokenzQdBNbLqP5VEhdkAS6EPFLC1PHnBqCXEpPxuEb",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "auth9SigNpDKz4sJJ1DfCTuZrZNSAgh9sFD3rboVmgg",
  },
  {
    "isSigner": false,
    "isWritable": false,
    "pubkey": "metaqbxxUerdq28cj1RbAWkYQm3ybzjb6a8bt518x1s",
  },
]
`;
